types/src/block_executor/config.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright © Aptos Foundation
+
+use crate::on_chain_config::BlockGasLimitType;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug)]
+pub struct BlockExecutorLocalConfig {
+    pub concurrency_level: usize,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct BlockExecutorOnchainConfig {
+    pub block_gas_limit_type: BlockGasLimitType,
+}
+
+impl BlockExecutorOnchainConfig {
+    pub fn new_no_block_limit() -> Self {
+        Self {
+            block_gas_limit_type: BlockGasLimitType::NoLimit,
+        }
+    }
+
+    pub fn new_maybe_block_limit(maybe_block_gas_limit: Option<u64>) -> Self {
+        Self {
+            block_gas_limit_type: maybe_block_gas_limit
+                .map_or(BlockGasLimitType::NoLimit, BlockGasLimitType::Limit),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct BlockExecutorConfig {
```

Comment:
Can this struct be combined with LocalConfig (just extract concurrency_level) and moved to `aptos-move` and used locally? AFAICT its just used within that crate. But I might have missed something.